OT-Redis (#1765)
SumoLogic/sumologic-documentation: docs/integrations/databases/open-telemetry/redis.md

> id: redis

Reviewer: Missing the three dashes above the frontmatter

```suggestion
---
id: redis
```

title: Sumo Logic Open Telemetry App for Redis
sidebar_label: Redis

> description: Documentation for Open Telemetry App for Redis.

Reviewer 2: Can you please write the description that should say about the content of this doc?

Reviewer 2: OpenTelemetry should be one word.

---

import useBaseUrl from '@docusaurus/useBaseUrl';
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

<img src={useBaseUrl('img/integrations/databases/redis.png')} alt="Thumbnail icon" width="50"/>

> [Redis](https://redis.io/docs/about/) is an in-memory data structure implementing a distributed, in-memory key-value database with optional durability.

Reviewer 2: I'd rephrase this as:
Redis is a distributed key-value database that operates in memory and has the option for durability. It implements a data structure that is stored in the memory of the computer and can be accessed quickly, making it useful for applications that require high performance.

> [Redis](https://redis.io/docs/about/) is an in-memory data structure implementing a distributed, in-memory key-value database with optional durability.

Reviewer 2: I'd rephrase this as:
[Redis](https://redis.io/docs/about/) is an in-memory data structure that implements a distributed, in-memory key-value database with durability options.

> The Sumo Logic App for Redis helps you monitor the Redis database cluster. The preconfigured dashboards provide detailed analysis based on logs. The panels provide details such as RDBMemory Usage, events, RDB and AOF events.

Reviewer 2: ```suggestion
The Sumo Logic App for Redis helps you monitor the Redis database cluster. The preconfigured dashboards provide detailed analysis based on logs. The panels provide details such as RDBMemory Usage, events, RDB, and AOF events.
```

Redis logs are sent to Sumo Logic through OpenTelemetry [filelog receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/filelogreceiver).

<img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Redis-OpenTelemetry/schematics-redis.png')} alt="Redis Schematics" />

> ##Redis Log Types

Reviewer 3: The heading is not appearing properly when I render it. I thing the problem is since there is no space between ## and the actual text.

> ##Redis Log Types

Reviewer: ```suggestion
## Redis Log Types
```

> This Sumo Logic App for Redis supports Logs for Redis in Cluster or Standalone mode. The Redis logs are generated in files as configured in the cluster node configuration files in `/etc/redis/redis.conf`. For more details on Redis logs, [click here](https://redislabs.com/ebook/part-2-core-concepts/chapter-5-using-redis-for-application-support/5-1-logging-to-redis/).

Reviewer 2: ```suggestion
This Sumo Logic App for Redis supports logs for Redis in Cluster or Standalone mode. The Redis logs are generated in files as configured in the cluster node configuration files in `/etc/redis/redis.conf`. For more details on Redis logs, [click here](https://redislabs.com/ebook/part-2-core-concepts/chapter-5-using-redis-for-application-support/5-1-logging-to-redis/).
```

> ##Fields creation in Sumo Logic for Redis

Reviewer 3: Same problem. Heading not appearing correctly.

> ##Fields creation in Sumo Logic for Redis

Reviewer 2: I'd say :
Creating Fields in Sumo Logic for Redis

> Following are the [Fields](https://help.sumologic.com/docs/manage/fields/), which will be created as part of Redis App install if not already present.

Reviewer 2: ```suggestion
Following are the [Fields](https://help.sumologic.com/docs/manage/fields/), which will be created as part of the Redis App install if not already present.
```

> **db.cluster.name** - User configured. Enter a name to identify this Redis cluster. This cluster name will be shown in the Sumo Logic dashboards.

Reviewer 2: I'd suggest we should create a table and use the single quotes for fields or parameters.
Like this. `db.cluster.name` 

Bold should be used when we talk about the UI elements.

**db.system** - Has fixed value of redis
**deployment.environment** - User configured. This is the deployment environment where the Redis cluster resides. For example: dev, prod or qa.
**sumo.datasource** - has fixed value of redis

##Prerequisite

> This section provides instructions for configuring log collection for Redis running on a non-Kubernetes environment for the Sumo Logic App for Redis. By default, Redis logs are stored in a log file.

Reviewer 2: ```suggestion
This section provides instructions for configuring log collection for Redis running on a non-Kubernetes environment for the Sumo Logic App for Redis. By default, Redis logs are stored in a log file.
```
```suggestion
This section provides instructions for configuring log collection for Redis running on a non-Kubernetes environment. By default, Redis logs are stored in a log file.
```

Follow the instructions to set up log collection:

1. To configure the Redis log file, locate your local `[redis.conf](https://download.redis.io/redis-stable/redis.conf)` configuration file in the database directory. By default, Redis logs are stored in `/var/log/redis/redis-server.log`.

> After determining the location of conf file modify the `redis.conf` configuration file logging parameters:

Reviewer 2: ```suggestion
After determining the location of the `conf file`, modify the `redis.conf` configuration file logging parameters:
```

Reviewer 2: This should point no. 2.

>   *   Open `redis.conf` configuration file in a text editor.

Reviewer 3: All the points are rendered to be in the same hierarchy. Could you please refer the gdoc for the point hierarchy

>   *   Open `redis.conf` configuration file in a text editor.

Reviewer 2: Point 3.

>   *   Set the following config parameters:

Reviewer 2: Can you please correct the numbering here? And these should have numbered bullets.

    *   Specify the server verbosity level. The value loglevel in conf file can be set to one of the following:

>       *   debug (a lot of information, useful for development/testing)

Reviewer 2: I think it's better if we remove this parenthesis, and define the terms, a one-line description is enough.

Reviewer 2: The same should be followed in 47, 48,49, 50

      *   verbose (many rarely useful info, but not a mess like the debug level)
      *   notice (moderately verbose, what you want in production probably)
      *   warning (only very important/critical messages are logged) loglevel notice
    *   Specify the log file name. Also the empty string can be used to force Redis to log on the standard output. Note that if you use the standard output for logging but daemonize, logs will be sent to `/dev/null` logfile

>   *   Save the `redis.conf` file and restart the redis server using command : `sudo  service redis-server restart`

Reviewer 2: ```suggestion
  *   Save the `redis.conf` file and restart the Redis server using the command: `sudo  service redis-server restart`
```

> Once the logs are configured to be written to a local file, follow the below step to configure collection in Sumo.

Reviewer 2: ```suggestion
Once the logs are configured to be written to a local file, follow the steps below to configure the collection in Sumo Logic.
```

> ##Collection Configuration & App installation

Reviewer 2: ```suggestion
## Collection Configuration & App installation
```

Reviewer 2: I'd rephrase the heading as:

## Configuring Collection 
I don't see App Installation in this doc. If we don't have the app installation steps, we should remove this from the heading.

> As part of setting up the collection process and app installation user can select the App from App Catalog and click Install App. Please follow the steps below:

Reviewer 2: ```suggestion
As part of setting up the collection process and app installation, you can select the App from **App Catalog** and click **Install App**.  Follow the steps below:
```

Reviewer 2: We avoid using "please" in the customer-facing user documentation.

> ### Step1: Set up Collector

Reviewer 2: ```suggestion
### Step 1: Set up Collector
```

> If you want to use an existing OpenTelemetry Collector, then this step can be skipped by selecting the option of using an existing Collector.

Reviewer 2: This should come in note.

> If you want to create a new Collector, please select the "Add a new Collector" option.

Reviewer 2: ```suggestion
If you want to create a new Collector, select the **Add a new Collector** option.
```

Reviewer 2: Numbering is missing..

Select the platform for which you want to install the Sumo OpenTelemetry Collector.

> This will generate a command that can be executed in the machine that needs monitoring. Once executed, it will install the Sumo Logic OpenTelemetry Collector agent.

Reviewer 2: ```suggestion
This will generate a command that can be executed in the machine that needs monitoring. Once executed, it will install the Sumo Logic OpenTelemetry Collector agent. <br/> <img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Redis-OpenTelemetry/collector-screenshot.png')} alt="Collector" />
```

> 

Reviewer 2: ```suggestion
```

Reviewer 2: Can you please remove the in-line extra spaces?

> <img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Redis-OpenTelemetry/collector-screenshot.png')} alt="Collector" />

Reviewer 2: You need to remove this, coz we have added the image along with the previous step.

> ###Step2: Configure integration

Reviewer 2: ```suggestion
### Step2: Configuring Integration
```

> Open Telemetry works with a [configuration](https://opentelemetry.io/docs/collector/configuration/) yaml file with all the details concerning the data that needs to be collected. For example, it specifies the location of a log file that is read and sent to Sumo Logic platform.

Reviewer 2: ```suggestion
OpenTelemetry works with a [configuration](https://opentelemetry.io/docs/collector/configuration/) yaml file with all the details concerning the data that needs to be collected. For example, it specifies the location of a log file that is read and sent to the Sumo Logic platform.
```

> In this step we will be configuring the yaml required for Redis Collection.

Reviewer 2: ```suggestion
In this step, we will be configuring the yaml required for Redis Collection.
```

The log file path configured to capture redis logs must be given here.

The files are typically located in `/var/log/redis/redis-server.log`. If you are using a customized path, check the `[redis.conf](https://download.redis.io/redis-stable/redis.conf)` file for this information.

> Click on the "Download YAML File" button to get the yaml file.

Reviewer 2: ```suggestion
Click on the **Download YAML File** button to get the yaml file.
```

> <img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Redis-OpenTelemetry/configuration-screenshot.png')} alt="Configuration" />

Reviewer 2: remove this

> ###Step3: Sending logs to Sumo

Reviewer 2: ```suggestion
### Step 3: Sending logs to Sumo Logic
```

> Once you have the yaml file downloaded in step 2, please follow the below steps based on your environment

Reviewer 2: ```suggestion
Once you have the yaml file downloaded in step 2, please follow the below steps based on your environment.
```

Reviewer 2: You should link Step 2 here.

> ####Linux

Reviewer 3: Same problem. Heading not appearing correctly.

> ####Linux

Reviewer 2: ```suggestion
#### Linux
```

> 1.  Copy the yaml at `/etc/otelcol-sumo/conf.d/` folder in the Redis instance which needs to be monitored.

Reviewer 2: ```suggestion
1.  Copy the yaml at **/etc/otelcol-sumo/conf.d/** folder in the Redis instance which needs to be monitored.
```

2.  Restart the otelcol-sumo process using the below command `sudo systemctl restart otelcol-sumo`

> ####Mac

Reviewer 3: Same problem. Heading not appearing correctly.

> ####Mac

Reviewer 2: ```suggestion
#### Mac
```

> 1.  Copy the yaml at `/etc/otelcol-sumo/conf.d/` folder in the Redis instance which needs to be monitored.

Reviewer 2: This is the folder name, it should be in bold.

2.  Restart the otelcol-sumo process using the command `otelcol-sumo --config /etc/otelcol-sumo/sumologic.yaml --conf "glob:/etc/otelcol-sumo/conf.d/*.yaml"`

> After the successful execution of the above command, Sumo will start receiving the data from your host machine.

Reviewer 2: I'd rephrase this as:
Sumo Logic will start receiving data from your host machine when the above command is successfully executed.

> This will install the app to your Sumo Logic Org. The app consists of Dashboards.

Reviewer 3: Please add "Press Next" at the starting of this line.

Results will be available after a while. It is important to note that each panel slowly and automatically fills with data matching the time range query received since the panel was created. Your data should appear in the dashboard within 20 minutes, and you will see complete graphs and maps.

### Sample Log Messages in Non-Kubernetes environments

> 5275:S 17 Mar 2021  19:13:38.138 * MASTER <-> REPLICA sync: Finished with success

Reviewer: please format as code

### Sample Query

This sample Query is from the Redis - Logs dashboard > Logs panel.

```
Query String
db.cluster.name=* sumo.datasource="redis"
| json auto maxdepth 1 nodrop
| if  (isEmpty(log), _raw, log)  as message
| count by message
| limit  100
| fields message
```

> ##Viewing Redis Dashboards

Reviewer: missing space after ##

> ##Viewing Redis Dashboards

Reviewer 2: ```suggestion
## Viewing Redis Dashboards
```
